src/components/Link/Link.tsx

```diff
@@ -21,6 +21,8 @@ export interface LinkProps {
   external?: boolean;
   /** Makes the link color the same as the current text color and adds an underline */
   monochrome?: boolean;
+  /** Adds text decoration underline to the link*/
+  underline?: boolean;
```

Comment:
This might be controversial but I think the underline link should be the default as it's the most accessible. Having a prop to remove the underline in specific use cases where the user understands it is interactive e.g. a list would be better. Would love to know others thoughts?

Reply:
I was wondering this as well. I think that's the right decision in the long run. It might be worth adding an underline by default and building Polaris for web to see how much it would be affected by that change. 

If it's quite a bit then we might want to save a change like that for v6 before we fully default to all links being underlined.

Reply:
+1 to "underline by default, allow removing as an option" as giving a better out the box behaviour. By giving a underline prop we just force the problem downstream and now every consumer has to remember to add the underline prop if they want to have better accessibility.

Agree that this feels like a pretty big design change and that it might be best saved for a major version.

Reply:
I would be happy to make this PR a part of v6 and make links underlined by default.